File: F#/Fsharp/Histogram.fs
namespace Fsharp

type Histogram =

    static member Hist(s: string) : string =
        "uwxz"
        |> Seq.map (fun letter -> letter, Seq.length (Seq.filter (fun c -> c = letter) s))
        |> Seq.filter (fun (_, num) -> num > 0)
        |> Seq.map (fun (letter, num) -> System.String.Format("{0}  {1,-5} {2}", letter, num, new string ('*', num)))
        |> String.concat "\n"

File: Java/src/codewars/Histogram.java
// https://www.codewars.com/kata/59f44c7bd4b36946fd000052/java

package codewars;

import java.util.ArrayList;
import java.util.LinkedHashMap;

public class Histogram
{
	public static String Hist(String s)
	{
		LinkedHashMap<Character, Integer> count = new LinkedHashMap<>();
		for (Character letter : s.toCharArray())
		{
			Integer val = count.get(letter);
			if (val == null) count.put(letter, 1);
			else count.replace(letter, val + 1);
		}

		ArrayList<String> res = new ArrayList<>();
		for (Character letter : "uwxz".toCharArray())
		{
			Integer val = count.get(letter);
			if (val != null && val > 0)
			{
				String num = String.format("%-6d", val);
				String stars = "*".repeat(val);
				res.add(String.format("%s  %s%s", letter, num, stars));
			}
		}
		return String.join("\n", res);
	}
}
